src/assets/scripts/client/AppController.js

```diff
@@ -279,7 +279,7 @@ export default class AppController {
     updateViewControls() {
         const { current: airport } = AirportController;
 
-        this.canvasController.canvas.dirty = true;
+        this._eventBus.trigger(EVENT.MARK_DIRTY_CANVAS);
```

Comment:
gotta love that event bus! this will be _SUCH_ a better way to deal with that sort of thing.